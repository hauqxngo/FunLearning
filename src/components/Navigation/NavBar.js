import React, { useContext, useState } from "react";
import UserContext from "../../UserContext";
import {
  Nav,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  NavItem,
  NavLink,
  Collapse,
} from "reactstrap";
import "./Navbar.css";

/** Shows up on every page.
 *
 * When user is logged in, shows Categories, Profile, & Logout tabs
 * Otherwise, shows Login & Sign Up tabs
 *
 * App -> NavBar
 */

const NavBar = ({ logout }) => {
  // toggle burger menu at small dimensions
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => {
    setIsOpen((open) => !open);
  };

  // authentication check
  const { currentUser } = useContext(UserContext);

  const loggedInNav = () => {
    return (
      <div>
        <Navbar color="success" dark expand="md" light>
          <NavbarBrand href="/">FUN LEARNING</NavbarBrand>
          <NavbarToggler onClick={toggle} />
          <Collapse isOpen={isOpen} navbar>
            <Nav className="ms-auto" navbar>
              <NavItem>
                <NavLink href="/">Home</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/categories">Categories</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/profile">
                  {currentUser.firstName}'s Profile
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/" onClick={logout}>
                  Logout
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  };
  const loggedOutNav = () => {
    return (
      <div>
        <Navbar color="success" dark expand="md" light>
          <NavbarBrand href="/">FUN LEARNING</NavbarBrand>
          <NavbarToggler onClick={toggle} />
          <Collapse isOpen={isOpen} navbar>
            <Nav className="ms-auto" navbar>
              <NavItem>
                <NavLink href="/">Home</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/#features">Features</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/#about">About</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/login">Login</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/signup">Sign Up</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  };

  return <div>{currentUser ? loggedInNav() : loggedOutNav()}</div>;
};

export default NavBar;
