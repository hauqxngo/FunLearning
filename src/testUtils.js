import React from "react";
import UserContext from "./UserContext";

const testUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  photo_url: null,
};

const UserProvider = ({ children, currentUser = testUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
