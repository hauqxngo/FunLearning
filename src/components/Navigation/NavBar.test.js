import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";
import { UserProvider } from "../../testUtils";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
