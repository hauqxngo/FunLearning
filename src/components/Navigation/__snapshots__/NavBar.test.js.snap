// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <nav
        class="navbar navbar-expand-md navbar-light navbar-dark bg-success"
      >
        <div
          class="container-fluid"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            FUN LEARNING
          </a>
          <button
            aria-label="Toggle navigation"
            class="navbar-toggler"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="ms-auto navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/categories"
                >
                  Categories
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/profile"
                >
                  's Profile
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <div>
      <nav
        class="navbar navbar-expand-md navbar-light navbar-dark bg-success"
      >
        <div
          class="container-fluid"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            FUN LEARNING
          </a>
          <button
            aria-label="Toggle navigation"
            class="navbar-toggler"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="ms-auto navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/#features"
                >
                  Features
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/#about"
                >
                  About
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/login"
                >
                  Login
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/signup"
                >
                  Sign Up
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
